app/pods/components/player-code-challenge/component.js

```diff
@@ -1,6 +1,162 @@
-import Component from '@ember/component';
-import { computed } from '@ember/object';
+import Component from '@ember/component'
+import { alias } from 'ember-decorators/object/computed'
+import { service } from 'ember-decorators/service'
+import { action } from 'ember-decorators/object'
+import { computed } from 'ember-decorators/object'
+import config from 'codingblocks-online/config/environment'
 
-export default Component.extend({
-    code: computed.alias('payload')
-});
+
+export default class CodeChallengeComponent extends Component {
+    @service api
+    @service ajax
+    @service hbApi
+    @service currentUser
+    @service pollboy
+    @alias ('payload') code
+    
+
+    sourceCode = ''
+    customInput = ''
+    customOutput = ''
+    runError = ''
+    language = 'cpp'
+    
+    pollId = null
+    maxPollCount = 10
+    pollCount = 0
+
+    @computed('sourceCode')
+    sourceCodeBase64 () {
+        return window.btoa(this.get('sourceCode'))
+    }
+
+    @computed('problemJsonApiPayload')
+    problem () {
+        const payload = this.get('problemJsonApiPayload')
+        return payload.data.attributes
+    }
+
+    @computed('problemJsonApiPayload')
+    codeStubs () {
+        const payload = this.get('problemJsonApiPayload')
+        const checkList = {
+            java: true,
+            cpp: true,
+            c: true,
+            py2: true,
+            csharp: true,
+            js: true
+        }
+        return payload.included.filter(stubs => {
+            if (checkList[stubs.attributes.language]) {
+                checkList[stubs.attributes.language] = false
+                return true
+            } else {
+                return false
+            }
+        }).map(stub => stub.attributes)
+    }
+
+    @computed('customInput')
+    customInputBase64 () {
+        return window.btoa(this.get('customInput'))
+    }
+
+    didReceiveAttrs () {
+        this._super(...arguments)
+        const code = this.get('code')
+        this.get('hbApi').request('problems', {
+            data: {
+                contest_id: code.get('hbContestId'),
+                problem_id: code.get('hbProblemId')
+            }
+        }).then(result => {
+            console.log(result)
+            this.set('problemJsonApiPayload', result)
+        }).catch(err => {
+            console.log(err)
+        })
+    }
+
+    @action
+    runCode () {
+        this.get('ajax').request('https://judge.cb.lk/api/submission', {
```

Comment:
remove this integration please. This is a synchronous call which will be deprecated soon. We don't want to block our judge for a request from amoeba particularly when everything else is working asynchronously.